solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/util/FlowReceiverContainer.java

```diff
@@ -190,6 +207,10 @@ public MessageContainer receive(Integer timeoutInMillis) throws JCSMPException,
 	 * @param messageContainer The message
 	 */
 	public void acknowledge(MessageContainer messageContainer) {
+		if (transacted) {
+			throw new UnsupportedOperationException("Transactions do not support message settlements");
```

Comment:
This may break things? I assume it is done to find all places where it can happen and make suer it doe snot happen, when this happens at runtime it won't be recoverable, right?

Reply:
Yeah, its prevent it being invoked for transaction use cases. If it is, then our tests _should_ capture the error and naturally fail.